### PR TITLE
App setup

### PR DESCRIPTION
**Pull request by wesonweb**
# Initial app setup

**Setting up the API**

- gets Node server running and installs express
- adds various node packages
- creates an MongoDB Atlas db and connects to it

**Routes, Model and Controller**

Should really have put these as a separate PR but it kind of just happened! 

- creates routes
- creates Plant model and schema
- adds methods for creating a plant and getting all plants

### CODE REVIEW

File: server/controllers/plantController.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      temperature,
+      position,
+      toxicity,
+      flowers,
+      problems
+    })
+    res.status(200).json(newPlant)
+  } catch (err) {
+    console.log(err)
+    return res.status(400).json({message: 'Could not create plant'})
```
Comment (lc512k) [inline server/controllers/plantController.js:40]: Good error handling and use of status codes throughout. well done

File: server/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "node server.js",
+    "dev": "nodemon server.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
```
Comment (lc512k) [inline server/package.json:13]: Awesome, this is a lot of the same stuff we use :)

File: server/routes/plants.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const { createPlant, getPlants } = require('../controllers/plantController')
+const express = require('express')
+const router = express.Router()
+
+// GET all plants
```
Comment (lc512k) [inline server/routes/plants.js:5]: Good practice to add these short comments, makes the file easy to scan

**Review by lc512k — APPROVED** [review]
Looks great, Wes! Neat file and folder structure, I can see the beginning of an MVC structure starting, which makes it easy to know what's going on. The code is neat and well organized. I felt I knew exactly where it's headed as I was reading it. You make good use of status codes and logging in your error handling, which would make this easy to debug. And your commits are well organized, this is really helpful for much larger PRs when your reviewer might need to go through some of them individually to review each functionality independently to help make sense of something really complex. 

Well done! Looking forward to the rest of the controller and to what you'll go with for the views :)

**Comment by wesonweb** [comment]
Hey Laura, thank you for reviewing and for the comments! Good to know I'm on the right track :-) I'll merge it to main.

**Pull request merged**